tests/e2e/cluster_downgrade_test.go

```diff
@@ -233,3 +291,35 @@ func getMembersAndKeys(t *testing.T, cc *e2e.EtcdctlV3) (*clientv3.MemberListRes
 
 	return members, kvs
 }
+
+func generateIdenticalVersions(clusterSize int, ver *semver.Version) []*version.Versions {
+	ret := make([]*version.Versions, clusterSize)
+
+	for i := range clusterSize {
+		ret[i] = &version.Versions{
+			Cluster: ver.String(),
+			Server:  ver.String(),
+			Storage: ver.String(),
+		}
+	}
+
+	return ret
+}
+
+func generatePartialCancellationVersions(clusterSize int, membersToChange []int, ver *semver.Version) []*version.Versions {
+	ret := make([]*version.Versions, clusterSize)
+
+	for i := range clusterSize {
+		ret[i] = &version.Versions{
+			Cluster: ver.String(),
+			Server:  e2e.OffsetMinor(ver, 1).String(),
+			Storage: "",
```

Comment:
Will continue to investigate this tomorrow

Reply:
Thanks to @siyuanfoundation, it's intended behavior for node version 3.5. Refer to the following comment https://github.com/etcd-io/etcd/blob/291353bc09e84e901409b4087fc50c93def04ed7/server/etcdserver/server.go#L193-L194